### PR TITLE
fix: remove rustls from wasm

### PR DESCRIPTION
**Pull request by Maan2003**

### CODE REVIEW

**Review by dpc — APPROVED** [review]

**Review by elsirion — DISMISSED** [review]

**Comment by elsirion** [comment]
Needs some conditional imports apparently :(

Screenshot 2023-04-15 at 10-51-22 fix remove rustls from wasm · fedimint_fedimint@9107920


Eventually we might just want to further split core or move tls to server to get rid of that feature flag. Actually, did you check if it might be easier to move tls to server?

File: fedimint-core/src/encoding/mod.rs [hunk 1]
```diff
@@ -5,6 +5,8 @@
 mod btc;
 mod secp256k1;
 mod tbs;
+
+#[cfg(not(target_wasm = "wasm"))]
```
Comment (Maan2003) [inline fedimint-core/src/encoding/mod.rs:9]: :smile: `target_wasm = "wasm"`

File: fedimint-core/src/lib.rs [hunk 2]
```diff
@@ -22,6 +22,7 @@ pub use crate::core::server;
 use crate::encoding::{Decodable, DecodeError, Encodable};
 use crate::module::registry::ModuleDecoderRegistry;
 
+#[cfg(not(target_family = "wasm"))]
```
Comment (elsirion) [inline fedimint-core/src/lib.rs:25]: Eventually I assume we'll also want the admin client to be WASM compatible? But I'm ok with getting it ready in a separate PR if it's too big.

**Review by elsirion — APPROVED** [review]

**Comment by Maan2003** [comment]
> did you check if it might be easier to move tls to server?

It is not, because we can only implement the Encodable trait for foreign types in fedimint-core

**Pull request merged**